test/sql/types/time/test_time_ns.test

```diff
@@ -0,0 +1,349 @@
+23:35:57.1234567	{'millisecond': 57123, 'microsecond': 57123456, 'epoch': 84957.123456}
+23:35:57.9999999	{'millisecond': 57999, 'microsecond': 57999999, 'epoch': 84957.999999}
+13:00:00.00000001	{'millisecond': 0, 'microsecond': 0, 'epoch': 46800.0}
+13:00:00.12345678	{'millisecond': 123, 'microsecond': 123456, 'epoch': 46800.123456}
+13:00:00.99999989	{'millisecond': 999, 'microsecond': 999999, 'epoch': 46800.999999}
+23:59:59.000000001	{'millisecond': 59000, 'microsecond': 59000000, 'epoch': 86399.0}
+23:59:59.123456789	{'millisecond': 59123, 'microsecond': 59123456, 'epoch': 86399.123456}
+23:59:59.999999999	{'millisecond': 59999, 'microsecond': 59999999, 'epoch': 86399.999999}
+24:00:00	{'millisecond': 0, 'microsecond': 0, 'epoch': 86400.0}
+NULL	NULL
+
+query II
+SELECT tns, DATE_PART(['timezone', 'timezone_hour', 'timezone_minute'], tns) p
+FROM times
+WHERE p <> {'timezone': 0, 'timezone_hour': 0, 'timezone_minute': 0}
+----
+
+# Invalid parts
+foreach part year month day quarter dow isodow doy week yearweek 
+
+statement error
+SELECT date_part('${part}', '23:59:59.123456789'::TIME_NS);
+----
+Binder Error: No function matches the given name and argument types
+
+statement error
+SELECT date_part(['${part}'], '23:59:59.123456789'::TIME_NS);
+----
+not recognized
+
+endloop
+
+foreach isoyear era
+
+statement error
+SELECT date_part('${part}', '23:59:59.123456789'::TIME_NS);
+----
+not recognized
+
+statement error
+SELECT date_part(['${part}'], '23:59:59.123456789'::TIME_NS);
+----
+not recognized
+
+endloop
+
+statement error
+SELECT date_part('julian', '23:59:59.123456789'::TIME_NS);
+----
+Binder Error: julian can only take DATE or TIMESTAMP arguments
+
+statement error
+SELECT date_part(['julian'], '23:59:59.123456789'::TIME_NS);
+----
+not recognized
+
+#
+# Casts
+#
+
+# TIME_NS => TIME
+query II
+SELECT tns, tns::TIME t
+FROM times
+----
+00:00:00	00:00:00
+00:01:20	00:01:20
+10:21:00	10:21:00
+10:21:00.1	10:21:00.1
+10:21:00.9	10:21:00.9
+16:04:22.01	16:04:22.01
+16:04:22.12	16:04:22.12
+16:04:22.97	16:04:22.97
+20:08:10.001	20:08:10.001
+20:08:10.123	20:08:10.123
+20:08:10.998	20:08:10.998
+03:45:47.0001	03:45:47.0001
+03:45:47.1234	03:45:47.1234
+03:45:47.9999	03:45:47.9999
+02:27:19.00001	02:27:19.00001
+02:27:19.12345	02:27:19.12345
+02:27:19.99899	02:27:19.99899
+09:01:54.000001	09:01:54.000001
+09:01:54.123456	09:01:54.123456
+09:01:54.999978	09:01:54.999978
+23:35:57.0000001	23:35:57
+23:35:57.1234567	23:35:57.123457
+23:35:57.9999999	23:35:58
+13:00:00.00000001	13:00:00
+13:00:00.12345678	13:00:00.123457
+13:00:00.99999989	13:00:01
+23:59:59.000000001	23:59:59
+23:59:59.123456789	23:59:59.123457
+23:59:59.999999999	24:00:00
+24:00:00	24:00:00
+NULL	NULL
+
+# TIMESTAMP_NS => TIME_NS
+query I
+SELECT '2025-05-20 15:30:00.123456789'::TIMESTAMP_NS::TIME_NS t;
+----
+15:30:00.123456789
+
+query I
+SELECT '1962-05-20 15:30:00.123456789'::TIMESTAMP_NS::TIME_NS t;
+----
+15:30:00.123456789
```

Comment:
I see that `CREATE SHOW TABLE` is not common in the test cases. Does the spec above support it?

Reply:
Ah, some sql flavors have that for printing the create table statement. I verified the show table …; works as expected